File: python_backend/api_helpers/convert_offset_strs_to_pairs.py
from pprint import pprint


def convert_offset_strs_to_pairs(offsets):
    """convert offsets from form 'x,y' to form [x,y] """
    for i in range(len(offsets)):
        offsets[i] = offsets[i].split(",")
        offsets[i][0] = int(offsets[i][0])
        offsets[i][1] = int(offsets[i][1])
    return offsets


if __name__ == "__main__":
    pprint(convert_offset_strs_to_pairs(["1,-1", "3,-2", "5,-8", "7,3"]))

File: app/src/apiHelpers/convertOffsetStrsToPairs.js


export function convertOffsetStrsToPairs(offsets) {
    /*convert offsets from form 'x,y' to form [x,y] **/
    var newOffsets = JSON.parse(JSON.stringify(offsets))
    for (let i = 0; i < newOffsets.length; i++) {
        newOffsets[i] = newOffsets[i].split(",")
        newOffsets[i][0] = parseInt(newOffsets[i][0], 10)
        newOffsets[i][1] = parseInt(newOffsets[i][1], 10)
    }
    return newOffsets;
}

// module.exports = convertOffsetStrsToPairs;

/** for node.js
if (require.main === module) {
    console.log(convertOffsetStrsToPairs(["1,-1", "3,-2", "5,-8", "7,3"]))

}
*/